server/api/student/student.controller.js

```diff
@@ -84,6 +87,123 @@ exports.outreachCounts = function(req, res) {
+          var records = this.records;
+          var typeArr = _.map(records, 
+              function(rec) { return rec.type 
+            });
+          return _.countBy(typeArr, _.identity);
+        }
+        School.findById(result.school, 
+          function(err, school) {
+            if (err) reject(err);
+            result.school = school;
+          })
+          .then(function() {
+            Student.findById(result.student, 
+              function(err, student) {
+              if (err) reject(err);
+              result.student = student;
+            })
+          .then(function() {
+            result.totals = getTotals.call(result);
+            resolve(result);
+          });
+        });
+      });
+    });
+    Promise.all(promises).then(
+      function(val) { 
+        resolve(val);
+    });
+  });
+}
+
+/**
+ * Get summary of outreaches.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get outreach summary for assignment school
+ * - manager+ will get outreach summary for all schools
+ */
+exports.outreachSummary = function(req, res) {
+  var pipeline = [{
+    $group: {
+      _id: {student: '$student', school: '$school'},
+      records: { $addToSet: '$$ROOT' }
+    } 
+  }, {
+    $project: {
+      _id: 0,
+      school: '$_id.school',
+      student: '$_id.student',
+      records: {
+        type: 1,
+        tier: 1,
+        absences: 1,
+        record: 1,
+        schoolYear: 1,
+        triggerDate: 1,
+        actionDate: 1
+      }
+    }
+  }];
+  if(req.user.role === 'teacher') {
+    pipeline.push({ 
+      $match: { school: req.user.assignment } 
+    });
+  }
+  Outreach.aggregate(pipeline, function(err, results) {
+    if (err) handleError(res, err);
+    var updated = createNameAndSchoolPromise(results);
+    updated.then(function(result) {
+      return res.status(200).json(result);
+    });
+  });
+};
+
+/**
+ * Get summary of interventions.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get intervention summary for assignment school
+ * - manager+ will get intervention summary for all schools
+ */
+exports.interventionSummary = function(req, res) {
+  var pipeline = [{
+    $group: {
+      _id: {student: '$student', school: '$school' },
+      records: { $push: '$$ROOT' }
+    } 
+  }, {
+    $project: {
+      _id: 0,
+      student: '$_id.student',
+      school: '$_id.school',
+      // Might need some fine tuning if implemented
+      records: 1
+    }
+  }];
+  if(req.user.role === 'teacher') {
+    pipeline.push({ 
+      $match: { school: req.user.assignment } 
+    });
+  }
+  Intervention.aggregate(pipeline, function(err, results) {
+    if (err) handleError(res, err);
+    var updated = createNameAndSchoolPromise(results);
+    updated.then(function(updated) {
+      return res.status(200).json(updated);
+    });
+  });
+};
+
```

Comment:
Didn't connect this, but thought it worthwhile to create if we need it down the road.